x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForSnapshotStep.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.xpack.core.slm.SnapshotLifecycleMetadata;
+import org.elasticsearch.xpack.core.slm.SnapshotLifecyclePolicyMetadata;
+
+import java.util.Objects;
+
+public class WaitForSnapshotStep extends ClusterStateWaitStep {
+
+    static final String NAME = "wait-for-snapshot";
+
+    private static final String MESSAGE_FIELD = "message";
+    private static final String POLICY_NOT_EXECUTED_MESSAGE = "waiting for policy '%s' to be executed";
+    private static final String POLICY_NOT_FOUND_MESSAGE = "policy '%s' not found, waiting for it to be created and executed";
+
+    private final String policy;
+
+    WaitForSnapshotStep(StepKey key, StepKey nextStepKey, String policy) {
+        super(key, nextStepKey);
+        this.policy = policy;
+    }
+
+    @Override
+    public Result isConditionMet(Index index, ClusterState clusterState) {
+        long phaseTime = LifecycleExecutionState.fromIndexMetadata(clusterState.metaData().index(index)).getPhaseTime();
+
+        SnapshotLifecycleMetadata snapMeta = clusterState.metaData().custom(SnapshotLifecycleMetadata.TYPE);
+        if (snapMeta == null || snapMeta.getSnapshotConfigurations().containsKey(policy) == false) {
+            return new Result(false, info(POLICY_NOT_FOUND_MESSAGE));
```

Comment:
I think rather than this never passing, it should signal in a better way, so I'm thinking that instead of return a `false` result, this should instead throw an exception and enter the ERROR step, but that we should then mark this step as retryable (override `isRetryable()` to return `true) so that it automatically retries every 10 minutes to see whether the policy exists.

Reply:
I'm not sure if we should use the ERROR step to signal conditions were not met (in time?). I think as this is an ILM *WaitStep (be it ClusterState or Async) it make sense to never pass if the condition is never met (I'd say it's the slm lifecycle's responsibility to signal it was not able to take another snapshot)

Reply:
Hmm... I can see what you're saying, but at the same time, this is still kind of an error since the policy was not found. I still think it would be preferable to signal via an `ERROR` (and retry) to make this more prominent (for example, if the user had a typo in the name of the policy).

Reply:
I agree with @dakrone here, most of the time SLM policy will be there when creating ILM policy, it's worth signaling to the user that there's potential error in his configuration. 
We do return `false` result when all required configuration can be retrieved and we can check if snapshot was taken during waiting, otherwise `ERROR` makes sense for me as this is not normal situation.
I've changed code accordingly.